projects/storefrontlib/src/cms-components/configurator/commons/config-previous-next-buttons/config-previous-next-buttons.component.ts

```diff
@@ -20,69 +21,92 @@ import { map, take } from 'rxjs/operators';
 })
 export class ConfigPreviousNextButtonsComponent implements OnInit {
   configuration$: Observable<Configurator.Configuration>;
-  productCode: string;
-  subscription = new Subscription();
+  ready$: Observable<boolean>;
 
   constructor(
     private routingService: RoutingService,
-    private configuratorGroupsService: ConfiguratorGroupsService
+    private configuratorGroupsService: ConfiguratorGroupsService,
+    private configuratorCommonsService: ConfiguratorCommonsService
   ) {}
 
   @Output() nextGroup = new EventEmitter();
   @Output() previousGroup = new EventEmitter();
 
   ngOnInit(): void {
-    this.subscription.add(
-      this.routingService
-        .getRouterState()
-        .subscribe(state => this.initConfigurationGroups(state))
+    this.configuration$ = this.routingService.getRouterState().pipe(
+      map(routingData => routingData.state.params.rootProduct),
+      switchMap(product =>
+        this.configuratorCommonsService.getConfiguration(product)
+      ),
+      tap(configuration => {
+        this.ready$ = this.configuratorCommonsService.isConfigurationReady(
```

Comment:
Similar to https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/5328/files#r345813019
Btw. are we using this stream `ready$` anywhere?

Reply:
No we don't :-)
This was work in progress, the component has been refactored and is a bit leaner. I pushed some logic to the facade layer and introduced new actions and effects